src/modules/roc_fec/reader.cpp

```diff
@@ -626,6 +579,157 @@ bool Reader::validate_sbn_sequence_(const packet::PacketPtr& pp) {
     return true;
 }
 
+bool Reader::check_payload_size_(size_t new_payload_size) {
+    if (payload_size_ == new_payload_size) {
+        return true;
+    }
+
+    if (payload_resized_) {
+        roc_log(LogDebug,
+                "fec reader: can't change payload size in the middle of a block:"
+                " next_esi=%lu cur_size=%lu new_size=%lu",
+                (unsigned long)next_packet_, (unsigned long)payload_size_,
+                (unsigned long)new_payload_size);
+        return false;
+    }
+
+    return true;
+}
+
+bool Reader::update_payload_size_(size_t new_payload_size) {
+    if (payload_size_ == new_payload_size) {
+        payload_resized_ = true;
+        return true;
+    }
+
+    roc_log(LogDebug,
+            "fec reader: update payload size: next_esi=%lu cur_size=%lu new_size=%lu",
+            (unsigned long)next_packet_, (unsigned long)payload_size_,
+            (unsigned long)new_payload_size);
+
+    payload_size_ = new_payload_size;
+    payload_resized_ = true;
+
+    return true;
+}
+
+bool Reader::check_source_block_size_(size_t new_sblen) {
+    const size_t cur_sblen = source_block_.size();
+
+    if (cur_sblen == new_sblen) {
+        return true;
+    }
+
+    if (source_block_resized_) {
+        roc_log(LogDebug,
+                "fec reader: can't change source block size in the middle of a block:"
+                " next_esi=%lu cur_sblen=%lu new_sblen=%lu",
+                (unsigned long)next_packet_, (unsigned long)cur_sblen,
+                (unsigned long)new_sblen);
+        return false;
+    }
+
+    if (new_sblen > decoder_.max_block_length()) {
```

Comment:
As we do not allow zero number of repair packets we should ensure that `new_sblen` also does not equal to `decoder_.max_block_length()`.

Reply:
This is already handled in validate_incoming_repair_packet_(). We have a test for this BTW.